packages/components/src/components/post-linkarea/post-linkarea.tsx

```diff
@@ -1,30 +1,49 @@
 import { Component, Element, h, Host } from '@stencil/core';
 import { version } from '@root/package.json';
 
+const INTERACTIVE_ELEMENTS = ['a'].join(',');
+const INTERACTIVE_ELEMENTS_SELECTOR = `:where(${INTERACTIVE_ELEMENTS})`;
+
+type InteractiveElement = HTMLAnchorElement;
+
 @Component({
   tag: 'post-linkarea',
   styleUrl: 'post-linkarea.scss',
 })
 export class PostLinkarea {
-  @Element() hostElement: HTMLPostLinkareaElement;
+  @Element() host: HTMLPostLinkareaElement;
 
-  private handleClick = () => {
-    const customSlottedLink: HTMLAnchorElement = this.hostElement.querySelector('a[data-link]');
+  private delegateClick() {
+    // get any child with a data-link attribute and check that it is interactive
+    const elementWithDataLink = this.host.querySelector('[data-link]');
+    if (!this.isInteractive(elementWithDataLink)) {
+      throw new Error(
+        `The \`data-link\` attribute must be used on a interactive element inside the \`post-linkarea\` component. Possible elements are: ${INTERACTIVE_ELEMENTS}`,
+      );
```

Comment:
I feel like we should not throw an error if there is at least one interactive element, even though it may not contain the [data-link] attribute. Instead I would log a warning and only throw an error if we really can't find any interactive element. Because then, and only then, the component can not work as expected.